lib/Conversion/ImportVerilog/Expressions.cpp

```diff
@@ -663,6 +663,24 @@ struct LvalueExprVisitor {
         lowBit);
   }
 
+  Value visit(const slang::ast::MemberAccessExpression &expr) {
+    auto type = context.convertType(*expr.type);
+    auto valueType = expr.value().type;
+    auto value = context.convertLvalueExpression(expr.value());
+    if (!type || !value)
+      return {};
+    if (valueType->isStruct()) {
+      return builder.create<moore::StructExtractRefOp>(
+          loc, moore::RefType::get(cast<moore::UnpackedType>(type)),
+          builder.getStringAttr(expr.member.name), value);
+    }
+    if (valueType->isPackedUnion() || valueType->isUnpackedUnion()) {
+      return builder.create<moore::UnionExtractRefOp>(
+          loc, moore::RefType::get(cast<moore::UnpackedType>(type)),
+          builder.getStringAttr(expr.member.name), value);
+    }
+    llvm_unreachable("unsupported symbol kind");
```

Comment:
Can you turn this into an error diagnostic since there doesn't seem to be a strict guarantee on Slang's side that it won't produce something beside `struct` and `union` types. (It's likely to produce these also for classes and mabye interfaces.)

Something like:
```c++
mlir::emitError(value.getLoc()) << "cannot access member";
return {};
```

Reply:
You are certainly right. 
But "error: unsupported construct: ClassType" this error will emitted firstly. Seems I can not add related error examples in errors.sv =(